internal/common/types.go

```diff
@@ -0,0 +1,126 @@
+package store
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/google/uuid"
+)
+
+// ParseCustomTypeError is thrown when parsing strings into enumerations
+type ParseCustomTypeError struct {
+	reason string
+}
+
+func (err *ParseCustomTypeError) Error() string {
+	return err.reason
+}
+
+// Architecture represents one of the supported CPU architectures available for images
+// produced by osbuild-composer. I decided to represent it as an integer because if it
+// was a string it would unmarshal from JSON just fine even in case that the architecture
+// was unknown (yeah, thanks Go for doing the same mistakes C did almost 50 years ago)
+//
+// I considered some alternatives like a struct with private fields, but struct cannot
+// be const, so that does not help either. Eventually I think this is the "get s**t done"
+// solution and next time please use a language with decent type system.
```

Comment:
This comment would be better suited to the commit message ("why I did what I did", as opposed to "how it works"). And comments about Go would be better suited to Twitter ;)

Reply:
Hehe, I don't use Twitter or any other social network, but I'll drop it if you insist!